extensions/common/vault/vault-hashicorp/src/main/java/org/eclipse/edc/vault/hashicorp/HashicorpVaultExtension.java

```diff
@@ -144,6 +147,7 @@ private HashicorpVaultSettings getSettings(ServiceExtensionContext context) {
         var ttl = context.getSetting(VAULT_TOKEN_TTL, VAULT_TOKEN_TTL_DEFAULT);
         var renewBuffer = context.getSetting(VAULT_TOKEN_RENEW_BUFFER, VAULT_TOKEN_RENEW_BUFFER_DEFAULT);
         var secretPath = context.getSetting(VAULT_API_SECRET_PATH, VAULT_API_SECRET_PATH_DEFAULT);
+        var folderPath = context.getSetting(VAULT_FOLDER_PATH, "");
```

Comment:
I'm not a fan of the empty string because it can lead to misunderstandings. 
I would set `null` as its default value, then in `HashicorpVaultClient` in the `getSecretUrl` method call the `addPathSegments` method with `folderPath` only if that's not null.

Reply:
:D But setting a proper default value instead of just `null` allows us to avoid any kind of nullpointer exceptions if someone forgets the null check.

Reply:
changed the default to null